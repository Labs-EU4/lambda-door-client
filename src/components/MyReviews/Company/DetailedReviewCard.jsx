/* eslint-disable no-shadow */
/* eslint-disable react/jsx-one-expression-per-line */
import React, { useState, useEffect } from 'react';
import {
  Rate,
  Switch,
  Icon,
  Card,
  Button,
  Skeleton,
  Popconfirm,
  Typography,
} from 'antd';
import styled from 'styled-components';
import { withRouter, useParams } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  deleteCompanyReview,
  updateCompanyReview,
} from '../../../state/actions/reviews';
import openNotification from '../../../utils/openNotification';
import { mobilePortrait, tabletPortrait } from '../../../styles/theme.styles';

const { Paragraph } = Typography;
let updatedReview;

export const DetailedReviewCard = ({
  history,
  reviews: {
    reviews: { company },
  },
  deleteCompanyReview,
  updateCompanyReview,
}) => {
  const [isEditing, setEditing] = useState(false);
  const [loading, setLoading] = useState(false);

  const reviewId = useParams().id;
  const review = company.find(elem => elem.id === Number(reviewId));

  useEffect(() => {
    updatedReview = { ...review };
    delete updatedReview.name;
    delete updatedReview.full_name;
    delete updatedReview.company_id;
    delete updatedReview.email_address;
  }, [review]);

  const handleDelete = async id => {
    await deleteCompanyReview(id);
    history.push(`/reviews/`);
    openNotification('Review deleted successfully!');
  };

  const updateReview = (key, value) => {
    updatedReview[key] = value;
  };

  const handleEdit = async () => {
    setLoading(true);
    await updateCompanyReview(updatedReview);
    setLoading(false);
    setEditing(false);
  };

  return !review ? (
    <Skeleton />
  ) : (
    <>
      <Button
        style={{
          marginBottom: '30px',
          border: '1px solid #BB1333',
          color: '#BB1333',
        }}
        onClick={() => history.push(`/reviews/`)}
      >
        <Icon type="left" />
        Back to Reviews
      </Button>
      <StyledReview>
        <div className="title-div">
          <h2>Company Name</h2>
          <span
            className="company"
            onClick={() => {
              history.push(`/company-page/${review.company_id}`);
            }}
          >
            {review.name}
          </span>
        </div>
        <div className="ratings">
          <h2>Overall Rating</h2>
          <div className="stars" data-testid={review.ratings}>
            <Rate
              disabled={!isEditing}
              defaultValue={review.ratings}
              onChange={e => updateReview('ratings', e)}
            />
          </div>
        </div>
        <div className="switch-statements">
          <div
            className="current-employee"
            data-testid={`employee - ${review.is_currently_employed}`}
          >
            <h2>I am a current employee</h2>
            <Switch
              checkedChildren={<Icon type="check" />}
              unCheckedChildren={<Icon type="close" />}
              defaultChecked={review.is_currently_employed}
              disabled={!isEditing}
              onChange={e => {
                updateReview('is_currently_employed', e === true ? 1 : 0);
              }}
            />
          </div>

          <div
            className="accepting-questions"
            data-testid={`questions - ${review.is_accepting_questions}`}
          >
            <h2>Accepting questions</h2>
            <Switch
              checkedChildren={<Icon type="check" />}
              unCheckedChildren={<Icon type="close" />}
              defaultChecked={review.is_accepting_questions}
              disabled={!isEditing}
              onChange={e => {
                updateReview('is_accepting_questions', e === true ? 1 : 0);
              }}
            />
          </div>
        </div>
        <div>
          <div className="headline-div">
            <h2>Review Headline </h2>
            <Paragraph
              editable={{
                onChange: e => {
                  updateReview('review_headline', e);
                },
                editing: isEditing,
              }}
              className="editable-text headline"
            >
              {review.review_headline}
            </Paragraph>
          </div>
          <div className="review-body">
            <h2>Review</h2>
            <Paragraph
              className="editable-text"
              editable={{
                onChange: e => {
                  updateReview('review', e);
                },
                editing: isEditing,
              }}
            >
              {review.review}
            </Paragraph>
          </div>
        </div>
        <div className="buttons">
          {!isEditing && (
            <Popconfirm
              placement="bottom"
              title="Are you sure to delete this review?"
              onConfirm={() => handleDelete(review.id)}
              okText="Yes"
              cancelText="No"
            >
              <Button type="danger" size="default">
                <Icon type="delete" /> Delete
              </Button>
            </Popconfirm>
          )}

          {isEditing ? (
            <Button
              style={{ backgroundColor: '#40A9FF', color: 'white' }}
              size="default"
              onClick={handleEdit}
              loading={loading}
              disabled={loading}
            >
              <span>
                <Icon type="save" /> Save
              </span>
            </Button>
          ) : (
            <Button
              style={{
                border: '1px solid #40A9FF',
                color: '#40A9FF',
              }}
              size="default"
              onClick={() => setEditing(!isEditing)}
            >
              <span>
                <Icon type="edit" /> Edit
              </span>
            </Button>
          )}

          {isEditing && (
            <Button
              size="default"
              type="danger"
              onClick={() => setEditing(false)}
            >
              <Icon type="close" /> Cancel
            </Button>
          )}
        </div>
      </StyledReview>
    </>
  );
};

const StyledReview = styled(Card)`
  max-width: 800px;
  padding: 50px !important;

  @media ${mobilePortrait} {
    padding: 0 !important;
  }

  @media ${tabletPortrait} {
    padding: 0 !important;
  }

  Paragraph {
    font-size: 20px;
  }
  .title-div {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    width: 70%;
    margin: 0;
    margin-bottom: 20px;
    span {
      margin-bottom: 0;
    }
    @media ${mobilePortrait} {
      flex-direction: column;
      justify-content: flex-start;
      align-items: flex-start;
      margin: 0;
      width: 100%;
      span {
        margin: 0;
        font-size: 16px;
      }
      h2 {
        margin: 0;
        font-size: 18px;
        width: 100%;
      }
    }
    @media ${tabletPortrait} {
      flex-direction: column;
      justify-content: flex-start;
      align-items: flex-start;

      margin: 0;
      width: 100%;
      span {
        margin: 0;
        font-size: 18px;
      }
      h2 {
        /* margin: 20px; */
        font-size: 20px;
        width: 50%;
      }
    }
  }
  .company {
    font-size: 1.2rem;
    margin-left: 42px;
    margin-bottom: 20px;
    cursor: pointer;
    transition: 1s hover;
    &:hover {
      opacity: 0.6;
    }
    &:active {
      transform: scale(1.05);
    }
  }
  .ratings {
    display: flex;
    width: 50%;
    justify-content: space-between;
    margin-bottom: 20px;
    @media ${mobilePortrait} {
      justify-content: flex-start;
      margin: 0;
      margin-top: 20px;
      width: 100%;

      .stars {
        transform: scale(0.8);
        width: 60%;
      }
      h2 {
        margin: 0;
        font-size: 18px;
        width: 50%;
      }
    }
    @media ${tabletPortrait} {
      justify-content: flex-start;
      margin: 0;
      margin-top: 20px;
      width: 100%;

      h2 {
        margin: 0;
        font-size: 20px;
        width: 50%;
      }
    }
  }
  .switch-statements {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-bottom: 20px;
    @media ${mobilePortrait} {
      flex-direction: column;
      margin-top: 20px;
    }
    @media ${tabletPortrait} {
      flex-direction: column;
      margin-top: 20px;
    }
    .current-employee {
      display: flex;
      width: 50%;
      justify-content: space-between;
      align-items: center;

      @media ${mobilePortrait} {
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 100%;

        h2 {
          margin: 0;
          font-size: 18px;
        }
      }

      @media ${tabletPortrait} {
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 90%;

        h2 {
          margin: 0;
          font-size: 20px;
        }
      }
    }
  }
  .accepting-questions {
    display: flex;
    width: 50%;
    justify-content: space-between;
    align-items: center;
    margin-top: 20px;
    @media ${mobilePortrait} {
      display: flex;
      justify-content: space-between;
      align-items: center;
      margin-left: 0;
      margin-top: 20px;
      width: 100%;

      h2 {
        margin: 0;
        font-size: 18px;
      }
    }
    @media ${tabletPortrait} {
      display: flex;
      justify-content: space-between;
      align-items: center;
      margin-top: 20px;
      margin-left: 0;
      width: 90%;

      h2 {
        margin: 0;
        font-size: 20px;
      }
    }
  }

  .headline-div {
    margin-bottom: 20px;
    display: flex;
    align-items: center;

    @media ${mobilePortrait} {
      flex-direction: column;
      align-items: flex-start;
      h2 {
        margin: 0;
        font-size: 18px;
        width: 100%;
      }
      .headline {
        margin-left: 0;
        font-size: 16px;
      }
    }
    @media ${tabletPortrait} {
      flex-direction: column;
      align-items: flex-start;

      h2 {
        margin: 0;
        font-size: 20px;
        width: 100%;
      }
      .headline {
        margin-left: 0;
        font-size: 18px;
      }
    }
    h2 {
      margin: 0;
    }
  }

  h2 {
    margin: 0;
  }

  .headline {
    font-size: 1.1rem;
    margin: 0;
    margin-left: 42px;
    width: 50%;
  }
  .review-body {
    @media ${mobilePortrait} {
      h2 {
        font-size: 18px;
        margin-bottom: 0;
      }
    }
  }

  .buttons {
    margin-top: 50px;
    display: flex;
    justify-content: space-between;
    width: 30%;
    @media ${mobilePortrait} {
      width: 100%;
      display: flex;
      justify-content: space-evenly;
    }
    @media ${tabletPortrait} {
      width: 100%;
      display: flex;
      justify-content: space-evenly;
    }
  }

  .editable-text {
    .ant-typography-edit {
      display: none !important;
    }
  }
`;

export default withRouter(
  connect(state => state, { deleteCompanyReview, updateCompanyReview })(
    DetailedReviewCard
  )
);
