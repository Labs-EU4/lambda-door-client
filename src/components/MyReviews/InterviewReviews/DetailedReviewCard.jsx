/* eslint-disable no-shadow */
/* eslint-disable react/jsx-one-expression-per-line */
import React, { useState, useEffect } from 'react';
import {
  Icon,
  Card,
  Button,
  Skeleton,
  Popconfirm,
  Typography,
  Switch,
} from 'antd';
import styled from 'styled-components';
import { withRouter, useParams } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  deleteInterviewReview,
  updateInterviewReview,
} from '../../../state/actions/reviews';
import openNotification from '../../../utils/openNotification';
import { mobilePortrait, tabletPortrait } from '../../../styles/theme.styles';

const { Paragraph } = Typography;
let updatedReview;

export const DetailedReviewCard = ({
  history,
  reviews: {
    reviews: { interview },
  },
  deleteInterviewReview,
  updateInterviewReview,
}) => {
  const [isEditing, setEditing] = useState(false);
  const [loading, setLoading] = useState(false);

  const reviewId = useParams().id;
  const review = interview.find(elem => elem.id === Number(reviewId));

  useEffect(() => {
    updatedReview = { ...review };
    delete updatedReview.full_name;
  }, [review]);

  const handleDelete = async id => {
    await deleteInterviewReview(id);
    history.push(`/reviews/`);
    openNotification('Review deleted successfully!');
  };

  const updateReview = (key, value) => {
    updatedReview[key] = value;
  };

  const handleEdit = async () => {
    setLoading(true);
    await updateInterviewReview(updatedReview);
    setLoading(false);
    setEditing(false);
  };

  return !review ? (
    <Skeleton />
  ) : (
    <>
      <Button
        style={{
          marginBottom: '30px',
          border: '1px solid #BB1333',
          color: '#BB1333',
        }}
        onClick={() => history.push(`/reviews/`)}
      >
        <Icon type="left" />
        Back to Reviews
      </Button>
      <StyledReview>
        <div className="title-div">
          <h2>Company Name</h2>
          <span className="company">{review.name}</span>
        </div>
        <div className="review-body">
          <h2>Review</h2>
          <Paragraph
            className="editable-text"
            editable={{
              onChange: e => {
                updateReview('text', e);
              },
              editing: isEditing,
            }}
          >
            {review.text}
          </Paragraph>
        </div>
        <div className="switch-statements">
          <div
            className="current-employee"
            data-testid={`employee - ${review.is_current_employee}`}
          >
            <h2>I am a current employee</h2>
            <Switch
              checkedChildren={<Icon type="check" />}
              unCheckedChildren={<Icon type="close" />}
              defaultChecked={review.is_current_employee}
              disabled={!isEditing}
              onChange={e => {
                updateReview('is_current_employee', e === true ? 1 : 0);
              }}
            />
          </div>

          <div
            className="accepting-questions"
            data-testid={`questions - ${review.is_accepting_questions}`}
          >
            <h2>Accepting questions</h2>
            <Switch
              checkedChildren={<Icon type="check" />}
              unCheckedChildren={<Icon type="close" />}
              defaultChecked={review.is_accepting_questions}
              disabled={!isEditing}
              onChange={e => {
                updateReview('is_accepting_questions', e === true ? 1 : 0);
              }}
            />
          </div>
        </div>
        <div className="buttons">
          {!isEditing && (
            <Popconfirm
              placement="bottom"
              title="Are you sure to delete this review?"
              okText="Yes"
              cancelText="No"
              onConfirm={() => handleDelete(review.id)}
            >
              <Button type="danger" size="default">
                <Icon type="delete" /> Delete
              </Button>
            </Popconfirm>
          )}

          {isEditing ? (
            <Button
              style={{ backgroundColor: '#40A9FF', color: 'white' }}
              size="default"
              onClick={handleEdit}
              loading={loading}
              disabled={loading}
            >
              <span>
                <Icon type="save" /> Save
              </span>
            </Button>
          ) : (
            <Button
              style={{
                border: '1px solid #40A9FF',
                color: '#40A9FF',
              }}
              size="default"
              onClick={() => setEditing(!isEditing)}
            >
              <span>
                <Icon type="edit" /> Edit
              </span>
            </Button>
          )}

          {isEditing && (
            <Button
              size="default"
              type="danger"
              onClick={() => setEditing(false)}
            >
              <Icon type="close" /> Cancel
            </Button>
          )}
        </div>
      </StyledReview>
    </>
  );
};

const StyledReview = styled(Card)`
  max-width: 800px;
  padding: 50px !important;

  @media ${mobilePortrait} {
    padding: 0 !important;
  }

  @media ${tabletPortrait} {
    padding: 0 !important;
  }

  Paragraph {
    font-size: 20px;
  }
  .title-div {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    width: 70%;
    margin: 0;
    margin-bottom: 20px;
    span {
      margin-bottom: 0;
    }
    @media ${mobilePortrait} {
      flex-direction: column;
      justify-content: flex-start;
      align-items: flex-start;
      margin: 0;
      width: 100%;
      span {
        margin: 0;
        font-size: 16px;
      }
      h2 {
        margin: 0;
        font-size: 18px;
        width: 100%;
      }
    }
    @media ${tabletPortrait} {
      flex-direction: column;
      justify-content: flex-start;
      align-items: flex-start;

      margin: 0;
      width: 100%;
      span {
        margin: 0;
        font-size: 18px;
      }
      h2 {
        /* margin: 20px; */
        font-size: 20px;
        width: 50%;
      }
    }
  }
  .company {
    font-size: 1.2rem;
    margin-left: 42px;
    margin-bottom: 20px;
  }
  .ratings {
    display: flex;
    width: 50%;
    justify-content: space-between;
    margin-bottom: 20px;
    @media ${mobilePortrait} {
      justify-content: flex-start;
      margin: 0;
      margin-top: 20px;
      width: 100%;

      .stars {
        transform: scale(0.8);
        width: 60%;
      }
      h2 {
        margin: 0;
        font-size: 18px;
        width: 50%;
      }
    }
    @media ${tabletPortrait} {
      justify-content: flex-start;
      margin: 0;
      margin-top: 20px;
      width: 100%;

      h2 {
        margin: 0;
        font-size: 20px;
        width: 50%;
      }
    }
  }
  .switch-statements {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-bottom: 20px;
    @media ${mobilePortrait} {
      flex-direction: column;
      margin-top: 20px;
    }
    @media ${tabletPortrait} {
      flex-direction: column;
      margin-top: 20px;
    }
    .current-employee {
      display: flex;
      width: 50%;
      justify-content: space-between;
      align-items: center;

      @media ${mobilePortrait} {
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 100%;

        h2 {
          margin: 0;
          font-size: 18px;
        }
      }

      @media ${tabletPortrait} {
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 90%;

        h2 {
          margin: 0;
          font-size: 20px;
        }
      }
    }
  }
  .accepting-questions {
    display: flex;
    width: 50%;
    justify-content: space-between;
    align-items: center;
    margin-top: 20px;
    @media ${mobilePortrait} {
      display: flex;
      justify-content: space-between;
      align-items: center;
      margin-left: 0;
      margin-top: 20px;
      width: 100%;

      h2 {
        margin: 0;
        font-size: 18px;
      }
    }
    @media ${tabletPortrait} {
      display: flex;
      justify-content: space-between;
      align-items: center;
      margin-top: 20px;
      margin-left: 0;
      width: 90%;

      h2 {
        margin: 0;
        font-size: 20px;
      }
    }
  }

  .headline-div {
    margin-bottom: 20px;
    display: flex;
    align-items: center;

    @media ${mobilePortrait} {
      flex-direction: column;
      align-items: flex-start;
      h2 {
        margin: 0;
        font-size: 18px;
        width: 100%;
      }
      .headline {
        margin-left: 0;
        font-size: 16px;
      }
    }
    @media ${tabletPortrait} {
      flex-direction: column;
      align-items: flex-start;

      h2 {
        margin: 0;
        font-size: 20px;
        width: 100%;
      }
      .headline {
        margin-left: 0;
        font-size: 18px;
      }
    }
    h2 {
      margin: 0;
    }
  }

  h2 {
    margin: 0;
  }

  .headline {
    font-size: 1.1rem;
    margin: 0;
    margin-left: 42px;
    width: 50%;
  }
  .review-body {
    @media ${mobilePortrait} {
      h2 {
        font-size: 18px;
        margin-bottom: 0;
      }
    }
  }

  .buttons {
    margin-top: 50px;
    display: flex;
    justify-content: space-between;
    width: 30%;
    @media ${mobilePortrait} {
      width: 100%;
      display: flex;
      justify-content: space-evenly;
    }
    @media ${tabletPortrait} {
      width: 100%;
      display: flex;
      justify-content: space-evenly;
    }
  }

  .editable-text {
    .ant-typography-edit {
      display: none !important;
    }
  }
`;

export default withRouter(
  connect(state => state, { deleteInterviewReview, updateInterviewReview })(
    DetailedReviewCard
  )
);
