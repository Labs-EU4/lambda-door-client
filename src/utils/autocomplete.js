/* eslint-disable react/jsx-no-undef */
import React, { useState } from 'react';
import { Form, Input, AutoComplete, Button } from 'antd';
import AddCompanyModal from '../components/AddReview/AddCompanyModal';

const { Option } = AutoComplete;

const renderOption = Item => {
  return (
    // eslint-disable-next-line react/jsx-filename-extension
    <Option key={Item.name} text={Item.name}>
      <p>{Item.name}</p>
    </Option>
  );
};

const AutoCompleted = ({ label, dataSource, placeholder, onChange }) => {
  const [addingCompany, setAddingCompany] = useState(false);
  const [value, setValue] = useState('');

  const options = dataSource.map(renderOption).concat([
    <Option key="all" value="" className="show-all">
      <Button type="button" onClick={() => setAddingCompany(true)}>
        Add Company
      </Button>
    </Option>,
  ]);

  return (
    <>
      <AddCompanyModal
        visible={addingCompany}
        setAddingCompany={setAddingCompany}
        value={value}
      />
      <Form.Item label={label}>
        <AutoComplete
          className="certain-category-search"
          dropdownClassName="certain-category-search-dropdown"
          dropdownMatchSelectWidth={false}
          dropdownStyle={{ width: 300 }}
          size="default"
          style={{ width: '100%' }}
          dataSource={options}
          optionLabelProp="value"
          filterOption={(inputValue, option) => {
            if (
              option.key.toLowerCase().includes(inputValue.toLowerCase()) ||
              option.key.toLowerCase() === 'all'
            ) {
              return true;
            }
            return false;
          }}
          onChange={e => {
            if (e) {
              onChange(e);
              setValue(e);
            }
          }}
        >
          <Input size="default" placeholder={placeholder} />
        </AutoComplete>
      </Form.Item>
    </>
  );
};

export default AutoCompleted;
