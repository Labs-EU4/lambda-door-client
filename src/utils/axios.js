import axios from 'axios';

axios.defaults.baseURL = `${process.env.REACT_APP_BACKEND_URL}`

export default function withAuth() {
  const token = localStorage.getItem('token');
  const instance = axios.create({
    headers: {
      'Content-Type': 'application/json',
      Authorization: `${token}`,
    },
  });

  return instance;
}
