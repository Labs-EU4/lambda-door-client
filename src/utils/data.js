export const companies = [
  {
    id: 1,
    name: 'Accenture',
    website: 'www.accenture.com.',
    location: 'Atlanta, GA',
    type: 'Business',
    logo: '',
    description: '',
  },
  {
    id: 2,
    name: 'Anthem, Inc.',
    website: 'https://www.antheminc.com',
    location: 'Atlanta, GA',
    type: 'Health',
    logo: '',
    description: '',
  },
  {
    id: 3,
    name: 'Ahalogy',
    website: 'https://www.ahalogy.com',
    location: 'Cincinnati, OH',
    type: 'Business',
    logo: '',
    description: '',
  },
  {
    id: 4,
    name: 'Bad Rabbit',
    website: 'https://www.badrabbit.com',
    location: 'Portland, OR',
    type: 'Technology',
    logo: '',
    description: '',
  },
  {
    id: 5,
    name: 'DoNotPay Inc',
    website: 'https://donotpay.com',
    location: 'Technology',
    type: 'rowValue3',
    logo: '',
    description: '',
  },
  {
    id: 6,
    name: 'Newfront Insurance',
    website: 'https://www.newfrontinsurance.com',
    location: 'San Francisco, CA',
    type: 'insurance',
    logo: '',
    description: '',
  },
  {
    id: 7,
    name: 'Unspun',
    website: 'https://www.unspuntech.com/',
    location: 'San Francisco, CA',
    type: 'Commerce',
    logo: '',
    description: '',
  },
  {
    id: 8,
    name: 'rideOS',
    website: 'https://rideos.ai',
    location: 'San Francisco, CA',
    type: 'Technology',
    logo: '',
    description: '',
  },
  {
    id: 9,
    name: 'Mesa',
    website: '',
    location: 'Austin, TX',
    type: 'Software',
    logo: '',
    description: '',
  },
  {
    id: 10,
    name: 'Paystack',
    website: 'https://paystack.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    description: '',
  },
  {
    id: 11,
    name: 'BuyCoins',
    website: 'https://buycoins.africa/',
    location: 'Lagos, NG',
    type: 'Crypto',
    logo: '',
    description: '',
  },
  {
    id: 12,
    name: 'Chipper Cash',
    website: 'https://chippercash.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    description: '',
  },
  {
    id: 13,
    name: 'Cowrywise',
    website: 'https://www.cowrywise.com/',
    location: 'Lagos, NG',
    type: 'FinTech',
    logo: '',
    description: '',
  },
  {
    id: 14,
    name: 'Others',
    website: '',
    location: '',
    type: '',
    logo: '',
    description: '',
  },
];
export const salaryReviews = [
  {
    id: 1,
    user_id: 1,
    company_id: 1,
    description: 'Accenture Programmer',
    salary: 95000,
    currency: 'USD',
    is_accepting_questions: 0,
  },
  {
    id: 2,
    user_id: 2,
    company_id: 3,
    description: 'Software Engineer',
    salary: 98000,
    currency: 'USD',
    is_accepting_questions: 0,
  },
  {
    id: 3,
    user_id: 3,
    company_id: 10,
    description: 'Junior Developer',
    salary: 3000000,
    currency: 'NGN',
    is_accepting_questions: 1,
  },
  {
    id: 4,
    user_id: 4,
    company_id: 1,
    description: 'Accenture Technical Specialist',
    salary: 85000,
    currency: 'USD',
    is_accepting_questions: 1,
  },
  {
    id: 5,
    user_id: 5,
    company_id: 13,
    description: 'Software Engineer',
    salary: '4800000',
    currency: 'NGN',
    is_accepting_questions: 0,
  },
];
export const interviewReviews = [
  {
    id: 1,
    user_id: 1,
    company_id: 1,
    company_name: 'Microsoft',
    text:
      'Six rounds of phone/tech interviews over a long time period. It seemed a bit scattered and could have been way more efficient. I felt like some of the interviews got repetitive. The entire process took three months and I was able to interview at another company and find a job whilst waiting to hear back.',
  },
  {
    id: 2,
    user_id: 2,
    company_id: 5,
    company_name: 'Google',
    text:
      'Very pleasant recruiters. First a phone interview that took about 15 minutes. Followed by two Skype interviews; first with a recruiter after that with a manager.Basic trouble shooting questions. They want to really get to know you as a person. Just be yourself and do not be afraid to be geeky/nerdy! Make sure you have a desk and dedicated home phone line.',
  },
  {
    id: 3,
    user_id: 4,
    company_id: 11,
    company_name: 'Amazon',
    text:
      'Completed their developer challenge and received an email offering an interview. A few preliminary questions via email. Scheduled a phone interview. The recruiter did not call me, and did not respond to my email asking why they missed the scheduled interview. I am not sure what happened, but after scheduling the interview, there is no reason to not show up.',
  },
  {
    id: 4,
    user_id: 3,
    company_id: 6,
    company_name: 'Twitter',
    text:
      'There is one phone interview for an hour. If you clear this interview, you will be invited for an onsite interview. There are five interviews onsite, all in one day. Interviews are in a casual environment. After the first two interviews, you take a break and are escorted for lunch.',
  },
  {
    id: 5,
    user_id: 5,
    company_id: 12,
    company_name: 'Slack',
    text:
      "They initially called me in for a technical interview and I didn't hear anything back for three weeks. After this I was surprised to get another call saying that I had been invited for a behavioural interview. In total the process took over two months. The interviews were both with a panel and were very thorough but the atmosphere was relaxed and friendly.",
  },

  {
    company_id: 21,
    company_name: 'Novosco',
    text:
      'I was invited to a very informal chat in the first place and shown around the office. Two weeks later I got a technical interview and I met some more people from the office. The interview lasted for an hour and had both easy and difficult questions. Two days later I got a phone call saying that I had got the job.',
  },
];

export const allInterests = [
  'Software Engineer',
  'Front End',
  'Back End',
  'Full Stack',
  'Data Science',
  'Machine Learning',
  'User Experience',
  'Mobile Development',
  'Product Manager',
];
export const jobCategories = [
  {
    id: 1,
    name: 'Software Engineer',
  },
  {
    id: 2,
    name: 'Front End',
  },
  {
    id: 3,
    name: 'Back End',
  },
  {
    id: 4,
    name: 'Full Stack',
  },
  {
    id: 5,
    name: 'Data Science',
  },
  {
    id: 6,
    name: 'Machine Learning',
  },
  {
    id: 7,
    name: 'User Experience',
  },
];
export const currencies = [
  {
    id: 1,
    name: 'USD',
    symbol: '$',
    full_name: 'USD Dollars',
  },
  {
    id: 2,
    name: 'EUR',
    symbol: '',
    full_name: 'EUROS',
  },
  {
    id: 3,
    name: 'GBP',
    symbol: '',
    full_name: 'British Pounds',
  },
  {
    id: 4,
    name: 'NAIRA',
    symbol: '',
    full_name: 'Nigerian Naira',
  },
];
