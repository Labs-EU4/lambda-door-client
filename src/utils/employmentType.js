export const employmentType = [
  {
    id: 1,
    name: 'Full-time',
  },
  {
    id: 2,
    name: 'Contract',
  },
  {
    id: 3,
    name: 'Part-time',
  },
  {
    id: 4,
    name: 'Internship',
  },
];
