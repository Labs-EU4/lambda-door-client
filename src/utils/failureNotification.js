import { notification } from 'antd';

const failureNotification = message => {
  notification.open({
    message: 'Something went wrong',
    description: message,
  });
};

export default failureNotification;
