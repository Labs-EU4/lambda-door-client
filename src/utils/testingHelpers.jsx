import React from 'react';
import { createStore, applyMiddleware, compose } from 'redux';
import { Provider } from 'react-redux';
import { Router, Route } from 'react-router-dom';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import rootReducer from '../state/reducers/index';

export const TestingRouter = ({
  ComponentWithRedirection,
  RedirectUrl,
  MainRoute,
}) => (
  <>
    <Route
      path={MainRoute}
      exact={true}
      render={props => {
        return <ComponentWithRedirection {...props} />;
      }}
    />
    <Route path={RedirectUrl} render={() => <div>{RedirectUrl}</div>} />
  </>
);

export const renderWithRedux = (
  component,
  {
    initialState,
    store = createStore(
      rootReducer,
      initialState,
      compose(applyMiddleware(thunk))
    ),
  } = {},
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>{component}</Router>
      </Provider>
    ),
    store,
    history,
  };
};
