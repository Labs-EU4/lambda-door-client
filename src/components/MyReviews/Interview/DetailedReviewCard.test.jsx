import React from 'react';
import * as rtl from '@testing-library/react';
import DetailedReviewCard from './DetailedReviewCard';
import '@testing-library/jest-dom/extend-expect';
import { renderWithRedux } from '../../../utils/testingHelpers';

beforeEach(rtl.cleanup);

describe('DetailedReviewCard', () => {
  it('renders without crashing', () => {
    renderWithRedux(<DetailedReviewCard />);
  });

  it('renders correctly', () => {
    expect(
      renderWithRedux(<DetailedReviewCard />).baseElement
    ).toMatchSnapshot();
  });
});