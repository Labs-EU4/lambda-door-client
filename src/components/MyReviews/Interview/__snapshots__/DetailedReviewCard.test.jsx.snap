// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedReviewCard renders correctly 1`] = `
<body>
  <div>
    <div
      class="ant-skeleton"
    >
      <div
        class="ant-skeleton-content"
      >
        <h3
          class="ant-skeleton-title"
          style="width: 38%;"
        />
        <ul
          class="ant-skeleton-paragraph"
        >
          <li />
          <li />
          <li
            style="width: 61%;"
          />
        </ul>
      </div>
    </div>
  </div>
</body>
`;
